lib/wab/impl/model.rb

```diff
@@ -20,10 +21,18 @@ class Model
       # dir:: directory to store data in
       def initialize(dir)
         @dir = dir
+        if dir.nil?
+          @dir = nil
+        else
+          @dir = ::File.expand_path(dir)
+        end
         @cnt = 0
```

Comment:
this entire method can be simplified into:
```ruby
      def initialize(dir)
        @dir = File.expand_path(dir) unless dir.nil?
        @cnt = 0
        @map = {}
        @lock = Thread::Mutex.new()

        Dir.mkdir(@dir) unless @dir.nil? && Dir.exist?(@dir)
        load_files()
      end
```

Reply:
I can see some of the changes but the lock still needs to be created.

Reply:
@dir also need to be set if dir is nil.

Reply:
lock?

Reply:
`@dir` is `nil` if undefined..

Reply:
Oh, I see it in th replacement. ok, it is needed.